MetaMask/metamask-extension
# Migrates avatar base to TypeScript

## Explanation

Currently, the AvatarBase component is built in JavaScript. This PR deprecates the JS component and creates a new TypeScript version.

Migrated all `.js` to `.tsx` or `.ts` files

- Converted objects to enum
- Updated storybook documentation as well as how to add a new icon
- Resolves #18063

## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/53999998/230484021-3b3eae39-4f8a-46b6-bc15-0580094a6a7a.mp4

### After

No changes to the functionality of the component, stories, and controls all work. Updated docs.

https://user-images.githubusercontent.com/53999998/230586371-0dd9978e-feba-4ac4-aed9-7974f9f5f230.mp4

## Manual Testing Steps

- Go to the storybook build on this PR
- Search AvatarBase in IDE and story in storybook.
- Check docs align with enums

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be required.
